gems/aws-crt-auth/lib/aws-crt-auth/signer.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Aws
+  module Crt
+    module Auth
+      # Utility class for creating AWS signature version 4 signature.
+      class Signer
+        def initialize(options = {})
+          @service = extract_service(options)
+          @region = extract_region(options)
+          @credentials_provider = extract_credentials_provider(options)
+          @unsigned_headers = Set.new((options.fetch(:unsigned_headers, []))
+                                      .map(&:downcase))
+          @unsigned_headers << 'authorization'
+          @unsigned_headers << 'x-amzn-trace-id'
+          @unsigned_headers << 'expect'
```

Comment:
I would avoid adding any "extra" functionality in the binding. Let the underlying C implementation be in charge. We want the CRTs to be consistent so we can fix issues in one place for every language.

The C implementation is already ignoring a bunch of headers by default. (:method, :scheme, :authority, :path, :status, cookie, host, connection, content-length, expect, transfer-encoding).

If you think it's annoying that this list is secret to the user, we can find some way to expose it.

Reply:
Agreed - I think this code will still have to exist in the SDK usage (see PR #21) - but the actual CRT bindings should remain un-opinionated/generic.  I've entirely removed this class from this PR.